Update gossipsub v1.1 spec with recommendations for network bootstrapping (#272)
libp2p/specs: pubsub/gossipsub/gossipsub-v1.1.md

  penalized by the score function. A peer transmitting lots of spam will quickly get graylisted,
  reducing the surface of spam-induced computation (eg validation). The application can take
  further steps and blacklist the peer if the spam persists after the negative score decays.

### Recommendations for Network Operators

An important issue to consider when deploying gossipsub is the peer discovery mechanism,
which must provide a secure way of discovering new peers.
Prior to gossipsub v1.1, operators were required to utilize an external peer discovery

> mechanism; with gossipsub v1.1 this is now entirely optional and the network can bootstrap

Reviewer: ```suggestion
mechanism to locate peers meshing in particular topics; with gossipsub v1.1 this is now entirely optional and the network can bootstrap
```

Author: the word "meshing" is probably not entirely correct, i think we need "participating".

Author: I'll commit and fix the word by hand.

> purely through a small set of network entry points (bootstrappers) by utilizing Peer Exchange.

Reviewer: ```suggestion
purely through a small set of network entry points (bootstrappers) by utilizing Peer Exchange. In other words, gossipsub 1.1 is now self-sufficient in this regard, as long as the node manages to find at least one peer meshing in the topic of interest.
```

Reviewer: ^^ is this correct, @vyzo?

Author: yeah, it's fine. Perhaps not using the word "meshing" though, as the bootstrappers are recommended to use `D=0` so they have no mesh.

In order to successfully bootstrap the network without a discovery service, network operators
should

> - Create and operate a set of bootstrapper nodes, that are known within client software.

Reviewer: ```suggestion
- Create and operate a set of stable bootstrapper nodes, whose addresses are known ahead of time by the application.
```

> - The bootstrappers should be configured without a mesh (ie set `D=Dlo=Dhi=Dout=0`)

Reviewer: Can we add an option to the pubsub constructor in the go reference implementation that does this, e.g. `Bootstrapper()`.

Author: yes, we can do that.

>   and with Peer Exchange enabled, utilizing Signed Peer Records.

Reviewer: We really need a proper spec for _signed peer records_ :-/

Reviewer 2: Very true. I can get that ready next week, I think. The implementation ended up tracking the RFC pretty closely, so it shouldn't take too much editing to make it into a spec.

> - Client software should assign a high application specific score to the bootstrappers and

Reviewer: ```suggestion
- The application should assign a high application-specific score to the bootstrappers and
```

>   set the `AcceptPXThreshold` to a high enough value attainable only by the bootstrappers.

Reviewer: ```suggestion
  set `AcceptPXThreshold` to a high enough value attainable only by the bootstrappers.
```

In this manner, the bootstrappers act purely as gossip and peer exchange nodes that facilitate
the formation and maintenance of the network.
Note that the score function is still present in the bootstrappers, which ensures that invalid
messages, colocation, and behavioural penalties apply to misbehaving nodes such that they do
not receive PX or are advertised to the rest of the network.

> In addition, network operators may configure the application specific scoring function such

Reviewer: ```suggestion
In addition, network operators may configure the application-specific scoring function such
```
